executor/new_distsql.go

```diff
@@ -189,10 +201,202 @@ func (e *IndexReaderExecutor) doRequest() error {
 	if err != nil {
 		return errors.Trace(err)
 	}
-	e.result, err = distsql.SelectDAG(e.ctx.GetClient(), goctx.Background(), e.dagPB, kvRanges, e.ctx.GetSessionVars().DistSQLScanConcurrency, e.keepOrder, e.desc)
+	e.result, err = distsql.SelectDAG(e.ctx.GetClient(), e.ctx.GoCtx(), e.dagPB, kvRanges, e.ctx.GetSessionVars().DistSQLScanConcurrency, e.keepOrder, e.desc)
 	if err != nil {
 		return errors.Trace(err)
 	}
 	e.result.Fetch(e.ctx.GoCtx())
 	return nil
 }
+
+// IndexLookUpExecutor implements double read for index scan.
+type IndexLookUpExecutor struct {
+	asName    *model.CIStr
+	table     table.Table
+	index     *model.IndexInfo
+	tableID   int64
+	keepOrder bool
+	desc      bool
+	ranges    []*types.IndexRange
+	dagPB     *tipb.DAGRequest
+	ctx       context.Context
+	schema    *expression.Schema
+
+	// result returns one or more distsql.PartialResult.
+	result distsql.SelectResult
```

Comment:
Well, both Index Reader and Table Reader have result, whose usage is reading data.